lib/rspec/rails/example/request_example_group.rb

```diff
@@ -1,5 +1,14 @@
 module RSpec
   module Rails
+    # Raises an error if the user tries to call integration test helpers.
+    # If the helpers are available then this module will be ignored.
+    module IntegrationHelperWarnings
+      extend ActiveSupport::Concern
+      [:assert_select, :url_for, :url_options].each do |name|
+        define_method(name) { |*args| raise "you must upgrade Rails to call #{name}" }
+      end
+    end
```

Comment:
Totally agree.  This is my way around Rubocop's "do not suppress exceptions error".  Turns out, after writing it, I actually like it.  Yay for helpful error messages.

I originally wrote this PR as a one-liner:
```ruby
include ActionDispatch::IntegrationTest::Behavior rescue NameError
```
but Rubocop didn't like that.

So I did a block:

```ruby
     begin
        include ActionDispatch::IntegrationTest::Behavior
      rescue NameError
        # if AD::IT::Behavior doesn't exist, that just means that Rails is old
        # and integration test helpers (like assert_select) aren't available.
      end
```

And it didn't like that either.  So I went all the way to this.

Happy to throw a `# rubocop:disable Lint/HandleExceptions` and got back to the one-liner you'd prefer.  But I kinda vote for the error messages.